core/connection.go

```diff
@@ -0,0 +1,119 @@
+package connection
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+/*
+This retry library was inspired by and uses Cenk Alti (https://github.com/cenkalti) backoff library (https://github.com/cenkalti/backoff).
+We would like to thank him for his great work.
+*/
+
+type PermanentError struct {
+	Inner error
+}
+
+func (e PermanentError) Error() string { return e.Inner.Error() }
+func (e PermanentError) Unwrap() error {
+	return e.Inner
+}
+func (e PermanentError) Is(err error) bool {
+	_, ok := err.(PermanentError)
+	return ok
+}
+
+type TransientError struct {
+	Inner error
+}
+
+func (e TransientError) Error() string { return e.Inner.Error() }
+func (e TransientError) Unwrap() error {
+	return e.Inner
+}
+func (e TransientError) Is(err error) bool {
```

Comment:
ditto.

Reply:
@Oppen any context regarding these implementations?. The original PR I branched off of included these functions for the Permanent error type. I implemented them following for the Transient error as well.

Reply:
transient error has been removed